Irara/modify gateway and client for goerli (#3)
NIC619/offchain-resolver: README.md

python3 -c "import os; import binascii; print('0x%s' % binascii.hexlify(os.urandom(32)).decode('utf-8'))"
```

Please set the private key to gateway .env file under `packages/gateway/`:

```bash
cp ./packages/gateway/.env.example ./packages/gateway/.env
code ./packages/gateway/.env
```

First, install dependencies and build all packages:

```bash
yarn clean && yarn install && yarn build
```

Next, run the gateway with the private key on the `Goerli` testnet:

```bash
yarn start-goerli:gateway --data token.eth.json
```

The value for the private key should be the key you set earlier in the .env file.

You will see output similar to the following:

```
yarn run v1.22.19
$ yarn workspace @ensdomains/offchain-resolver-gateway start-goerli --data token.eth.json
$ eval $(grep '^PRIVATE_KEY' .env) && node dist/index.js --private-key ${PRIVATE_KEY} --data token.eth.json

Serving on port 8080 with signing address 0x3B7D34d0E7e807A9D7aD74F094C5379aca61460D
```

Take a look at the data in `token.eth.json` under `packages/gateway/`; it specifies addresses for the name `token.eth` and the wildcard `*.token.eth`.

> Next, edit `packages/contracts/deployments/goerli/AddressRecord.json`; replacing the address of `Signer` and `Owner` with the one output when you ran the command above.

Reviewer: nit:
```suggestion
Next, edit `packages/contracts/deployments/goerli/AddressRecord.json`; replacing the address of `Signer` and `Owner` with the one output when you ran the command above (`0x3B7D...`).
```

Author: No problem! Thanks!

> And, `in a new terminal`, edit contracts .env file under `packages/contracts/`:

Reviewer: ```suggestion
And, in a new terminal, edit contracts .env file under `packages/contracts/`:
```

Author: No problem! Thanks!

```bash
cp ./packages/contracts/.env.example ./packages/contracts/.env
code ./packages/contracts/.env
```

> Then, connect to `Goerli` testnet with an ENS registry and the offchain resolver deployed:

Reviewer: ```suggestion
Then, connect to `Goerli` testnet and deploy ENS registry and the offchain resolver:
```

Author: No problem! Thanks!

```bash
yarn start-goerli:gateway --data token.eth.json
```

The value for the private key should be the key you set earlier in the .env file.

You will see output similar to the following:

```
Serving on port 8000 with signing address 0x3B7D34d0E7e807A9D7aD74F094C5379aca61460D
```

Take a look at the data in `token.eth.json` under `packages/gateway/`; it specifies addresses for the name `token.eth` and the wildcard `*.token.eth`.

Next, edit `packages/contracts/scripts/goerli/deployments/AddressRecord.json`; replacing the address of `Signer` and `Owner` with the one output when you ran the command above.

And, edit contracts .env file under `packages/contracts/`:

```bash
cp ./packages/contracts/.env.example ./packages/contracts/.env
code ./packages/contracts/.env
```

Then, in a new terminal, connect to `Goerli` testnet with an ENS registry and the offchain resolver deployed:

```bash
cd ./packages/contracts
npx hardhat run ./scripts/deploy/ENSRegistry.ts --network goerli
npx hardhat run ./scripts/deploy/OffchainResolver.ts --network goerli
```

Before setting the domain owner, move the new `ABI` files to `deployments` folder:

```bash
mkdir -p ./deployments/goerli

> mv -f ./scripts/goerli/deployments/ENSRegistry.json ./deployments/goerli/ENSRegistry.json

Reviewer: I think we can do it in next PR to write the deployment info directly to `./deployments` so others can skip this step?

Author: I think because on the "Hardhat" network executing a simulated deployment contract in LABS's tokenlon-contract-private Repo will permanently overwrite the abi files in "Mainnet" folder,
it will be manually overwritten by the user.

Not sure if it's safer to keep this traditional way in our offchain-resolver Repo?
